### ISSUE TITLE
BUG specifiying a list of categories to use fails at init time

### ISSUE THREAD

**Issue by pierreglaser** [dirty-cat/dirty_cat#86]
Specifiying a list of categories to use when creating a `SimilarityEncoder` fails:

```python
from dirty_cat import SimilarityEncoder
SimilarityEncoder(categories=['foo', 'bar'])
```

```python
AssertionError                            Traceback (most recent call last)
<ipython-input-2-713e2faae33f> in <module>
----> 1 SimilarityEncoder(categories=['foo', 'bar'])

~/repos/dirty_cat/dirty_cat/similarity_encoder.py in __init__(self, similarity, ngram_range, categories, dtype, handle_unknown, hashing_dim, n_prototypes, random_state)
    193         self.random_state = random_state
    194 
--> 195         assert categories in [None, 'auto', 'k-means', 'most_frequent']
    196         if categories in ['k-means', 'most_frequent'] and (n_prototypes is None or n_prototypes == 0):
    197             raise ValueError('n_prototypes expected None or a positive non null integer')

AssertionError: 
```
Apart from the obvious fix, will look for any further errors before submitting a PR.

**Comment by GaelVaroquaux** [comment]
@pierreglaser : is this still open?

**Comment by pierreglaser** [comment]
@GaelVaroquaux closing as fixed by #87
